File: solutions/javascript/bob/2/bob.js
(function() {
  "use strict";
  var Bob = (function() {
    function Bob() {}

    Bob.prototype.hey = function(str) {
      var returnString = 'Whatever.';
      if (str === '') {
        returnString = 'Fine, be that way!';
      } else if (str === str.toUpperCase()) {
        returnString = 'Woah, chill out!';
      } else if (str.slice(-1) === '?') {
        returnString = 'Sure.';
      }
      return returnString;
    };

    return Bob;
  })();

  module.exports = Bob;
}).call(this);

File: solutions/elixir/bob/2/bob.exs
defmodule Teenager do
  def hey(str) do
    cond do
      is_nothing?(str) -> "Fine. Be that way!"
      is_yelling?(str) -> "Woah, chill out!"
      is_asking?(str)  -> "Sure."
      true             -> "Whatever."
    end
  end

  defp is_yelling?(str), do: String.upcase(str) == str
  defp is_asking?(str),  do: String.last(str) == "?"
  defp is_nothing?(str) when str == nil, do: true
  defp is_nothing?(str), do: String.strip(str) == ""
end